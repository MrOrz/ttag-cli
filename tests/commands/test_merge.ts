import * as path from "path";
import { execSync } from "child_process";

const poPaths = [
    path.resolve(__dirname, "../fixtures/mergeTest/mergeLeft.po"),
    path.resolve(__dirname, "../fixtures/mergeTest/mergeRight.po"),
    path.resolve(__dirname, "../fixtures/mergeTest/mergeRightRight.po")
].join(" ");

test("merge two files together", () => {
    const result = execSync(`ts-node src/index.ts merge ${poPaths}`);
    const expectStr = expect(result.toString());
    expectStr.toContain('msgstr "test translated"');
    expectStr.toMatchSnapshot();
});
