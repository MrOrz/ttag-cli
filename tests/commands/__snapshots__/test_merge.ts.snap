// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merge two files together 1`] = `
"msgid \\"\\"
msgstr \\"\\"
\\"Content-Type: text/plain; charset=iso-8859-1\\\\n\\"
\\"Plural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"

#: tests/fixtures/checkTest/check-trans-exist.js:3
msgctxt \\"ctx\\"
msgid \\"test\\"
msgstr \\"test translated\\""
`;
